ui/src/app/pages/demographic-survey.tsx

```diff
@@ -60,7 +60,7 @@ export const DemographicSurvey = (props: WithSpinnerOverlayProps) => {
     await profileApi().updateProfile(profile);
     await profileStore.get().reload();
     hideSpinner();
-    navigateByUrl('/profile');
+    history.goBack();
```

Comment:
reminder to think about the cases where it does not make sense to goBack()
* user typed the url directly
* user was directed from email (hypothetical)

Reply:
I was not able to distinguish or find where the user is coming from, unless we start adding state in our application, which i think will be big task.

So i was thinking worse case: if i am using the application url directly it will redirect me back, in case of emails maybe we can prevent this by opening the survey in a new tab perhaps?

Reply:
I tested out the new tab scenario - goBack redirects the user to an empty page.
So I think `history.goBack()` won't work, I'm afraid.

How about this:
* Clicking the "take survey" button includes a param with the **source page location**
* **Submit** redirects to the page in the param if it exists, else /profile

Does this make sense?  It's kind of like what we're doing in modules-for-annual-renewal L282, `navigateByUrl('profile', { queryParams: { renewal: 1 } })`